### PR TITLE
Remove unnecessary overrides from inlineExpressions

### PR DESCRIPTION
**Pull request by viliam-durina**
The overrides were the same as the superclass impl.

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java [hunk 1]
```diff
@@ -482,40 +478,6 @@ public static RexNode inlineExpression(List<RexNode> inlinedExpressions, RexNode
...
-                return call.clone(call.type, newOperands);
-            }
-
-            @Override
-            public RexNode visitDynamicParam(RexDynamicParam dynamicParam) {
-                return dynamicParam;
-            }
-
-            @Override
-            public RexNode visitFieldAccess(RexFieldAccess fieldAccess) {
```
Comment (krzysztofslusarski) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java:506]: This method on superclass has a bit different semantic

```java
  @Override public RexNode visitFieldAccess(RexFieldAccess fieldAccess) {
    RexNode before = fieldAccess.getReferenceExpr();
    RexNode after = before.accept(this);

    if (before == after) {
      return fieldAccess;
    } else {
      return new RexFieldAccess(
          after,
          fieldAccess.getField());
    }
  }
```

original code throws ```RuntimeException```, the code of superclass does not. Won't this change break anything?

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java [hunk 2]
```diff
@@ -482,40 +478,6 @@ public static RexNode inlineExpression(List<RexNode> inlinedExpressions, RexNode
...
                 return inlinedExpressions.get(inputRef.getIndex());
             }
-
-            @Override
-            public RexNode visitLocalRef(RexLocalRef localRef) {
-                return localRef;
-            }
-
-            @Override
-            public RexNode visitCall(RexCall call) {
```
Comment (krzysztofslusarski) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java:492]: Superclass method is a bit different:

```java
  @Override public RexNode visitCall(final RexCall call) {
    boolean[] update = {false};
    List<RexNode> clonedOperands = visitList(call.operands, update);
    if (update[0]) {
      // REVIEW jvs 8-Mar-2005:  This doesn't take into account
      // the fact that a rewrite may have changed the result type.
      // To do that, we would need to take a RexBuilder and
      // watch out for special operators like CAST and NEW where
      // the type is embedded in the original call.
      return call.clone(call.getType(), clonedOperands);
    } else {
      return call;
    }
  }
```

The original code always create a clone, the code of a superclass does not. Is it ok?

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java (hunk 1 above)
Comment (viliam-durina) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java:506]: In this case the original code was actually broken. We hit the issue with @ivanthescientist with nested fields. In master code, we currently never use `RexFieldAccess`, so it wasn't visible.

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java (hunk 2 above)
Comment (viliam-durina) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java:492]: I think it is, it's even better.

**Review by krzysztofslusarski — APPROVED** [review]

**Review by ivanthescientist — APPROVED** [review]

**Pull request merged**